eng/configure-toolset.ps1

```diff
@@ -1,8 +1,13 @@
-# We depend on a local cli for a number of our buildtool
-# commands like init-tools so for now we need to disable
-# using the globally installed dotnet
-
-$script:useInstalledDotNetCli = $false
-
 # Working around issue https://github.com/dotnet/arcade/issues/2673
 $script:DisableNativeToolsetInstalls = $true
+
+function DotnetInstall()
+{
+  $repoRoot = $env:__RepoRootDir
+  $coreClrRoot = $env:__ProjectDir
+  $toolsScript = Join-Path $repoRoot "eng\common\tools.ps1"
+  . $toolsScript
+  $targetScript = Join-Path $coreClrRoot "bin\obj\set-dotnet-install-dir.cmd"
+  InitializeBuildTool
+  Set-Content -Path $targetScript -Value "set `"DOTNET_INSTALL_DIR=$env:DOTNET_INSTALL_DIR`""
```

Comment:
Does this work when the bin\obj folder doesn't exist yet? I've tried 
```ps1
Set-Content -Path "boo\moo\woo\log.txt" -Value "Hello"
```
And got an error:
```
Set-Content : Could not find a part of the path 'C:\Users\janvorli\boo\moo\woo\log.txt'.
```

Reply:
I was thinking about the transfer of output from powershell to batch more and seeing the DotnetInstall function, it seems you could pipe the output of the InitializeBuildTool to out-null and then return the $env:DOTNET_INSTALL_DIR. In this case, the env var content would be the only thing that gets to the stdout. So you can then capture it to a temp file and read the file content into the env var on the batch side. Like this:

Powershell:
```ps1
function DotnetInstall()
{
  $repoRoot = $env:__RepoRootDir
  $coreClrRoot = $env:__ProjectDir
  $toolsScript = Join-Path $repoRoot "eng\common\tools.ps1"
  . $toolsScript
  $targetScript = Join-Path $coreClrRoot "bin\obj\set-dotnet-install-dir.cmd"
  InitializeBuildTool | out-null
  return $env:DOTNET_INSTALL_DIR
}
```
bat:
```bat
for /f "delims=" %%l in ('powershell -NoProfile -ExecutionPolicy unrestricted -Command %PS_DOTNET_INSTALL_SCRIPT%') do set "DOTNET_INSTALL_DIR=%%l"
```

Reply:
Fixed in 2nd commit via a combination of your and Viktor's suggestions.